src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs

```diff
@@ -2803,68 +2803,56 @@ public string CreateEscapedIdentifier(string name)
         }
 
         // returns the type name without any array declaration.
-        private string GetBaseTypeOutput(CodeTypeReference typeRef)
+        private string GetBaseTypeOutput(CodeTypeReference typeRef, bool preferBuiltInTypes = true)
         {
             string s = typeRef.BaseType;
-            if (s.Length == 0)
-            {
-                s = "void";
-                return s;
-            }
 
-            string lowerCaseString = s.ToLower(CultureInfo.InvariantCulture).Trim();
+            if (preferBuiltInTypes)
+            {
+                if (s.Length == 0)
+                {
+                    return "void";
+                }
+
+                string lowerCaseString = s.ToLower(CultureInfo.InvariantCulture).Trim();
+
+                switch (lowerCaseString)
+                {
+                    case "system.int16":
+                        return "short";
+                    case "system.int32":
+                        return "int";
+                    case "system.int64":
+                        return "long";
+                    case "system.string":
+                        return "string";
+                    case "system.object":
+                        return "object";
+                    case "system.boolean":
+                        return "bool";
+                    case "system.void":
+                        return "void";
+                    case "system.char":
+                        return "char";
+                    case "system.byte":
+                        return "byte";
+                    case "system.uint16":
+                        return "ushort";
+                    case "system.uint32":
+                        return "uint";
+                    case "system.uint64":
+                        return "ulong";
+                    case "system.sbyte":
+                        return "sbyte";
+                    case "system.single":
+                        return "float";
+                    case "system.double":
+                        return "double";
+                    case "system.decimal":
+                        return "decimal";
+                }
```

Comment:
The diff showed the default case been removed. Saw the full file now, makes sense. The diff was confusing.